Core/GDCore/IDE/Events/EventsVariableReplacer.cpp

```diff
@@ -197,10 +205,11 @@ class GD_CORE_API ExpressionVariableReplacer
     // (and if it's a variable reference or a value does not have any importance
     // here).
 
-    if (forcedInitialVariablesContainer) {
+    if (forcedVariablesContainer) {
       // A scope was forced. Honor it: it means this node represents a variable
       // of the forced variables container.
-      if (forcedInitialVariablesContainer == &targetVariablesContainer) {
+      if (forcedVariablesContainer == &targetVariablesContainer ||
+          (forcedObjectName == targetGroupName && !targetGroupName.empty())) {
```

Comment:
All of these `xxx == targetGroupName && !targetGroupName.empty()` would almost mandate a `IsTheTargetGroupName(xxx)` helper function.

For the reader, it's hard to parse (especially because you check for targetGroupName not empty at the end, which is counter intuitive in my opinion). Let's make a helper so that it's easier to read the logic.